source/distributed/fully_distributed_tria.cc

```diff
@@ -539,35 +540,37 @@ namespace parallel
         dealii::Utilities::pack(construction_data, buffer, false);
 
         // Write offsets to file.
-        unsigned int buffer_size = buffer.size();
+        std::uint64_t buffer_size = buffer.size();
 
-        unsigned int offset = 0;
+        std::uint64_t offset = 0;
 
         ierr = MPI_Exscan(&buffer_size,
                           &offset,
                           1,
-                          MPI_UNSIGNED,
+                          MPI_UINT64_T,
                           MPI_SUM,
                           this->mpi_communicator);
         AssertThrowMPI(ierr);
 
         // Write offsets to file.
         ierr = MPI_File_write_at(fh,
-                                 myrank * sizeof(unsigned int),
+                                 myrank * sizeof(std::uint64_t),
                                  &buffer_size,
                                  1,
-                                 MPI_UNSIGNED,
+                                 MPI_UINT64_T,
                                  MPI_STATUS_IGNORE);
         AssertThrowMPI(ierr);
 
+        std::uint64_t global_position =
+          mpisize * sizeof(std::uint64_t) + offset;
         // Write buffers to file.
-        ierr = MPI_File_write_at(fh,
-                                 mpisize * sizeof(unsigned int) +
-                                   offset, // global position in file
-                                 buffer.data(),
-                                 buffer.size(), // local buffer
-                                 MPI_CHAR,
-                                 MPI_STATUS_IGNORE);
+        ierr = dealii::Utilities::MPI::LargeCount::MPI_File_write_at_c(
+          fh,
+          global_position, // global position in file
```

Comment:
The comment here can now go. That is why I suggested the new variable, by the way. Comment through code (variable names) is much cleaner than what was done here.